db/error_handler.cc

```diff
@@ -466,10 +466,13 @@ void ErrorHandler::SetBGError(const Status& bg_status,
     Status bg_err(new_bg_io_err, severity);
     CheckAndSetRecoveryAndBGError(bg_err);
     recover_context_ = context;
+    bool auto_recovery = true;
```

Comment:
From here: https://github.com/facebook/rocksdb/blob/3db030d7ee1b887ce818ec6f6a8d10949f9e9a22/db/error_handler.cc#L345-L357

`auto_recovery` is used as an output parameter to guide `ErrorHandler` for either to take actions or not, which maps the documentation in the event listener here https://github.com/facebook/rocksdb/blob/3db030d7ee1b887ce818ec6f6a8d10949f9e9a22/include/rocksdb/listener.h#L810
Should we also follow the same pattern to allow user to disable `ErrorHandler` auto resume in these cases and let the user to take over to manually resume. Maybe we can update the documentation too if we decide to change it.

Reply:
> Should we also follow the same pattern to allow user to disable ErrorHandler auto resume in these cases and let the user to take over to manually resume. 

Not sure yet. But right now the document should be updated to exclude those cases from being controlled by users cuz effectively only HandleKnownErrors() does that.

Regardless that seems to be irrelevant to this PR change though I wouldn't mind updating the API to spell out the restriction - did I miss something?

Reply:
The issue mentioned seems to be tangent to this PR change though.
```
if (*auto_recovery) {
      listener->OnErrorRecoveryBegin(reason, *bg_error, auto_recovery);
}
```
first relies on the auto_recovery value passed in to call OnErrorRecoveryBegin and then gives users the change to change it.

> Should we also follow the same pattern to allow user to disable ErrorHandler auto resume in these cases

Maybe - not sure if it's a feature incompletion or intentional left-out. Regardless, it's not a blocking issue to this PR though I won't mind updating the API doc to exclude some scenarios.

Reply:
@anand1976 Do you by any chance know why we didn't implement  https://github.com/facebook/rocksdb/commit/3db030d7ee1b887ce818ec6f6a8d10949f9e9a22 for flush or write back error recovery though the listener says https://github.com/facebook/rocksdb/blob/3db030d7ee1b887ce818ec6f6a8d10949f9e9a22/include/rocksdb/listener.h#L810

Reply:
I don't recall whether it was intentional or oversight. But its a good point. Right now it only seems to apply to the `kNoSpace/kSpaceLimit` class of errors. It makes sense to be consistent and make it applicable in all scenarios.

I agree its tangential to this PR though. We can tackle it in a separate PR.